### PR TITLE
dappeteer test for createPoolPage

### PR DESCRIPTION
**Pull request by smartdevcom**

### CODE REVIEW

**Comment by peetzweg** [comment]
Does this work? Haven't checked it in more detail yet. How was the experience, should we use it? :)

**Comment by smartdevcom** [comment]
Yes, it's much better than cypress-metamask-v2 or Synpress. closed to done of testing `CreatePoolPage`

File: packages/ui/.env.dist [hunk 1]
```diff
@@ -16,3 +16,5 @@ SUPABASE_URL=https://xdjnvsfkwtkwfuayzmtm.supabase.co
 SUPABASE_PLUGIN_TABLE_NAME=apy-plugin
 SUPABASE_FLYWHEEL_TABLE_NAME=apy-flywheel
 SUPABASE_NATIVE_PRICES_TABLE_NAME=native-prices
+TEST_SEED=
```
Comment (peetzweg) [inline packages/ui/.env.dist:19]: To what values do I need to set these?

File: packages/ui/.env.dist [hunk 2]
```diff
@@ -16,3 +16,5 @@ SUPABASE_URL=https://xdjnvsfkwtkwfuayzmtm.supabase.co
...
+TEST_PASS=
```
Comment (peetzweg) [inline packages/ui/.env.dist:20]: To what values do I need to set these?

**Comment by peetzweg** [comment]
So I just run?
`yarn install`

`yarn test:ui` 

Can we run them in CI as well?

**Comment by smartdevcom** [comment]
I think we should add it in CI check

**Comment by peetzweg** [comment]
This is great stuff. I adapted the create pool test a bit to work with BSC local fork, to not create more and more testnet pools on chapel.

Creation is a great thing to test, we should also test other admin function.

However, to get a feel we could just start with checking that data is actually loading in, like balances and pool information.

Can we rewrite the MIMO test to work for example with the BSC jarvis version? SO all tests run on BSC Fork for now?

@carlomazzaferro mentioned he has issues with the Polygon fork becoming out of sync fast.

**Comment by smartdevcom** [comment]
Okay, I will rewrite MIMO pool test in BSC forked

**Comment by smartdevcom** [comment]
After I deposited some bnb on BSC forked, supplied value doesn't get updated so test couldn't be passed. And also creating a pool is not working for me, Did they work on your side? cc @peetzweg

File: docker-compose.yaml [hunk 3]
```diff
@@ -7,7 +7,8 @@ services:
       context: .
     environment:
       - MNEMONIC
-      - FORK_URL_BSC
+      - FORK_URL_BSC # Leaving this in here, to potentially not break things?
```
Comment (peetzweg) [inline docker-compose.yaml:10]: @carlomazzaferro can you check this. I renamed this variable, maybe we need to adapt github secrets or other deployment environment variables somewhere?

File: packages/sdk/src/MidasSdk/index.ts [hunk 4]
```diff
@@ -98,11 +100,45 @@ export class MidasBase {
     return this._signer;
   }
 
+  public get contracts(): {
```
Comment (carlomazzaferro) [inline packages/sdk/src/MidasSdk/index.ts:103]: This will likely require changing all the bots code, AFAICT

Comment (carlomazzaferro) [reply packages/sdk/src/MidasSdk/index.ts:103]: What's the reasoning here btw?

**Comment by carlomazzaferro** [comment]
This is really dope. Just note my comment -- it'll require changing a bit the tests and the code for accessing the contracts, as that was refactored. Otherwise, looks really really awesome.

**Comment by carlomazzaferro** [comment]
@peetzweg : this is the only place where the refactoring is needed: https://github.com/Midas-Protocol/monorepo/blob/development/.github/workflows/package-sdk-pull-request-build-and-test.yml#L67

**Review by carlomazzaferro — APPROVED** [review]
Pending the Mocha tests and the refactorings needed to fix those, LGTM

File: packages/sdk/src/MidasSdk/index.ts (hunk 4 above)
Comment (peetzweg) [reply packages/sdk/src/MidasSdk/index.ts:103]: probably not, it's a getter function. `sdk.contract.FusePoolLens` works as before, no worries.

I had to do it as somehow the instance created in the constructor gets out of sync and is not updated. I don't like this implementation as well, but it solves this issue. Wasn't able to find a better solution in a reasonable amount of time.

**Pull request merged**